### PR TITLE
Create post

### PR DESCRIPTION
**Pull request by mobatra**

### CODE REVIEW

**Review by Nour-Agha99 — APPROVED** [review]

File: src/components/MyCohort/AddPost/index.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        } else {
+          await postsRoutes.create({
+            isPublic: true,
+            content: data.postContent,
+            image: null,
+          });
+        }
+        setSelectedImage(null);
+
+        toast.success('');
```
Comment (karamzomlot) [inline src/components/MyCohort/AddPost/index.tsx]: I think no need for this line.

File: src/components/MyCohort/AddPost/index.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      onError: (err: ReqError) => {
+        const errorMessage = err.response.data.data.message;
+        toast.error(errorMessage);
+      },
+    },
+  );
+
+  const handlePost = (data: FormData) => {
+    mutate(data);
+    reset();
```
Comment (karamzomlot) [inline src/components/MyCohort/AddPost/index.tsx]: If we reset the form values in `onSuccess` will be good
move it after line `79`

File: src/utils/validation/postSchema.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    .nonempty('Post cannot be empty')
+    .min(2, 'Post must be at least 2 characters long'),
+  postImage: z
+    .string()
+    .refine((value) => /\.(jpe?g|png|gif)$/gi.test(value), {
+      message: 'Invalid image format. Only JPEG, PNG, and GIF are supported.',
+    })
+    .optional()
+    .nullable(),
+});
```
Comment (karamzomlot) [inline src/utils/validation/postSchema.ts]: I think the validation doesn't work.
Coz I try to upload `svg` file and it was created

**Review by karamzomlot — CHANGES_REQUESTED** [review]
Delete any empty file (style.ts, index.tsx, etc...)

File: src/components/MyCohort/AddPost/index.tsx (hunk 2 above)
Comment (mobatra) [reply src/components/MyCohort/AddPost/index.tsx]: well, you want if there is an error don't reset the form ?

**Review by KhaledAref13 — APPROVED** [review]

**Review by Shatha-Amin — APPROVED** [review]

File: src/components/MyCohort/AddPost/index.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+            });
+          } else {
+            await postsRoutes.create({
+              isPublic: true,
+              content: data.postContent,
+              image: null,
+            });
+          }
+          setSelectedImage(null);
+          toast.success('Post created successfully');
```
Comment (karamzomlot) [inline src/components/MyCohort/AddPost/index.tsx]: Should be in `onSuccess` statement

File: src/utils/validation/postSchema.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { z } from 'zod';
+
+const postSchema = z.object({
+  postContent: z
+    .string()
+    .nonempty('Post cannot be empty')
+    .min(2, 'Post must be at least 2 characters long'),
+  postImage: z
```
Comment (karamzomlot) [inline src/utils/validation/postSchema.ts]: If you didn't use it delete it

**Review by karamzomlot — CHANGES_REQUESTED** [review]

**Review by karamzomlot — APPROVED** [review]

**Pull request merged**